docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -23,72 +23,106 @@ includes a boot loader.
+`/boot/` was traditionally handled), all installed OSes use the same place for
+boot loader menu entries._
+
+For systems where the firmware is able to read file systems directly, the ESP
+must — and the MBR boot and GPT XBOOTLDR partition should — be a file system
+readable by the firmware. For most systems this means VFAT (16 or 32
+bit). Applications accessing both partitions should hence not assume that
+fancier file system features such as symlinks, hardlinks, access control or
+case sensitivity are supported.
+
+### The `$BOOT` Partition Placeholder
+
+In the text below, the placeholder `$BOOT` will be used to refer to the
+partition determined as follows:
+
+ 1. On disks with an MBR partition table: → the boot partition, as described above
+
+ 2. On disks with a GPT partition table: → the XBOOTLDR partition if it exists
+
+ 3. Otherwise, on disks with a GPT partition table: → the ESP
+
+`$BOOT` is the *primary* place to put boot menu entry resources into, but
+typically not the only one. Most importantly, boot loaders should also pick up
+menu entries from the ESP, even if XBOOTLDR exists (for details see below).
+
+### Creating These Partitions
+
+An installer for an operating system should use this logic when selecting or
 creating partitions:
 
-  * If `$BOOT` is not found, a new suitably sized partition (let's say 500MB)
-    should be created, matching the characteristics described above. On disks
-    with GPT, only the ESP partition without the XBOOTLDR partition should be
-    created.
+  * If a boot partition (in case of MBR) or an XBOOTLDR partition (in case of
+    GPT) already exists it should be used as `$BOOT` and used as primary
+    location to place boot loader menu resources in.
 
-  * If the OS is installed on a disk with GPT and the ESP partition is found
-    but is too small, a new suitably sized (let's say 500MB) XBOOTLDR partition
-    shall be created.
+  * Otherwise, if on GPT and an ESP is found and it is large enough (let's say
+    at least 1G) it should be used as `$BOOT` and used as primary location to
+    place boot loader menu resources in.
 
-Those file systems shall be determined during _installation time_, and an fstab
-entry may be created. If only one partition is used, it should be mounted on
-`/boot/`. If both XBOOTLDR partition and the ESP are used, they should be
-mounted on `/boot` and `/efi`, or on `/boot` and `/boot/efi`.
+  * Otherwise, if on GPT and neither XBOOTLDR nor ESP exist, an ESP should be
+    created of the appropriate size and be used as `$BOOT`, and used as primary
+    location to place boot loader menu resources in.
 
-**Note:** _Those file systems are **shared** among all OS installations on the
-system. Instead of maintaining one boot partition per installed OS (as `/boot/`
-was traditionally handled), all installed OSes use the same place for boot-time
-configuration._
+  * Otherwise, a boot partition (in case of MBR) or XBOOTLDR partition (in case
```

Comment:
I'm confused. I thought $XBOOTLDR applies to MBR with 0xEA type code and GPT. What is a "boot partition" ? Seems vague, but also is not $XBOOTLDR.